### PR TITLE
🐛Prevents text overflow on share button in different languages on stories

### LINKED ISSUE #17631: Desktop share button label overflows
The desktop share button label overflows in some languages.
The container has a fixed width that doesn't work for all languages.

image

### PR DESCRIPTION
**Pull request by addieachan**
-Fixes #17631

### CODE REVIEW

**Comment by addieachan** [comment]
Video demo of it working: https://screencast.googleplex.com/cast/NTUyMDQ5Mjc4NjA4OTk4NHwyOTcyMTc4Yy1hNA

**Comment by newmuis** [comment]
Overall, this feels expensive; I don't think I get, conceptually, why a pure CSS solution isn't possible here?

**Comment by addieachan** [comment]
@gmajoulet @newmuis @Enriqe Ping (maybe <-- still trying to find CSS only a solution, but would like to hear feedback to my comment)

**Comment by newmuis** [comment]
After discussing with @gmajoulet I personally feel like this might be fine.  Hardcoding a width might be problematic since it won't scale up when e.g. a font fails to load and a wider font gets used as a fallback.  It's also probably okay since the calculation can happen once at build time.

File: extensions/amp-story/1.0/amp-story-system-layer.js [hunk 1]
```diff
@@ -564,11 +578,68 @@ export class SystemLayer {
...
+  }
+  /**
+   * Builds and appends the share pill. Desktop only.
+   * @private
+   */
+  changeMargin_() {
+    if (this.shareLabelWidth_ == null) {
+      const shareList = this.systemLayerEl_
+          .querySelector('.i-amphtml-story-share-list');
+      const elem = this.systemLayerEl_
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story-system-layer.js]: maybe change `elem` to something more descriptive? e.g. `pillLabelEl`?

File: extensions/amp-story/1.0/amp-story-system-layer.js [hunk 2]
```diff
@@ -564,11 +578,68 @@ export class SystemLayer {
...
+          .querySelector('.i-amphtml-story-share-pill-label');
+      const style = this.win_.getComputedStyle(elem);
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story-system-layer.js]: same with `style`, maybe `pillLabelStyle`

File: extensions/amp-story/1.0/amp-story-system-layer.js [hunk 3]
```diff
@@ -564,11 +578,68 @@ export class SystemLayer {
...
+      } else {
+        setImportantStyles(shareList, {'margin-right': px(margin)});
+      }      
+    }
+
+    const background = this.systemLayerEl_
+        .querySelector('.i-amphtml-story-share-pill-background');
+    setImportantStyles(background, {
+      'width': 'calc(100% - 32px)',
+      'max-width': px(800),
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story-system-layer.js]: maybe make `800` a const var at the top `MAX_WIDTH`

**Comment by newmuis** [comment]
We're going to see if there are other ways of addressing the share pill, since it's causing more problems than just this.  We can reopen this PR if we decide to go this route.

**Pull request closed without merging**